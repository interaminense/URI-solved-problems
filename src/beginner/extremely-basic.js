/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 * 
 * Read 2 variables, named A and B and make the sum of these two variables,
 * assigning its result to the variable X. Print X as shown below. Print
 * endline after the result otherwise you will get “Presentation Error”.
 * 
 * INPUT
 * 
 * The input file will contain 2 integer numbers.
 * @param {number} a
 * @param {number} b
 * 
 * OUTPUT
 * 
 * Print the letter X (uppercase) with a blank space before and after the equal
 * signal followed by the value of X, according to the following example.
 */
const sum = (a, b) => tmpl(a + b);

/**
 * Template definition
 * @param {string | number} result 
 */
const tmpl = result => `X = ${result}`;

export {sum, tmpl};