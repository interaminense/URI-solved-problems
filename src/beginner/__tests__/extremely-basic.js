import { sum, tmpl } from '../extremely-basic';

describe('Extremely basic', () => {
  let A = 0;
  let B = 0;
  const result = (a, b) => sum(a, b);

  it('The sum 10 + 9 must be 19', () => {
    A = 10;
    B = 9;

    expect(result(A, B)).toEqual(tmpl(19));
    expect(result(A, B)).toMatchSnapshot();
  });

  it('The sum -10 + 4 must be -6', () => {
    A = -10;
    B = 4;

    expect(result(A, B)).toEqual(tmpl(-6));
    expect(result(A, B)).toMatchSnapshot();
  });

  it('The sum 15 + (-7) must be 8', () => {
    A = 15;
    B = -7;

    expect(result(A, B)).toEqual(tmpl(8));
    expect(result(A, B)).toMatchSnapshot();
  });
});
