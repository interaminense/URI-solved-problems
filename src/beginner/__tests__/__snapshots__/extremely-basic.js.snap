// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Extremely basic The sum -10 + 4 must be -6 1`] = `"X = -6"`;

exports[`Extremely basic The sum 10 + 9 must be 19 1`] = `"X = 19"`;

exports[`Extremely basic The sum 15 + (-7) must be 8 1`] = `"X = 8"`;
